core/pom.xml

```diff
@@ -616,6 +621,50 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <configuration>
+          <shadedArtifactAttached>false</shadedArtifactAttached>
+          <shadeTestJar>true</shadeTestJar>
+          <artifactSet>
+            <includes>
+              <include>com.google.protobuf:*</include>
+            </includes>
+          </artifactSet>
+          <relocations>
+            <relocation>
+              <pattern>com.google.protobuf</pattern>
+              <shadedPattern>${spark.shade.packageName}.spark-core.protobuf</shadedPattern>
+              <includes>
+                <include>com.google.protobuf.**</include>
+              </includes>
+            </relocation>
+          </relocations>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>com.github.os72</groupId>
+        <artifactId>protoc-jar-maven-plugin</artifactId>
+        <version>3.11.4</version>
```

Comment:
shall we pick `3.21.9` as it was used in the protobuf module?

Reply:
Need to check the impact on hadoop-2, although Spark 3.4 will not release hadoop-2 distribution, but CI is still running

Reply:
This is shaded, I don't think the version matters to hadoop2?

Reply:
The plugin is already the latest: 
https://mvnrepository.com/artifact/com.github.os72/protoc-jar-maven-plugin

But yes we unify the versions with the one in protobuf connector.